examples/decoding/plot_decoding_ssd_csp_eeg.py

```diff
@@ -0,0 +1,256 @@
+"""
+# Authors: Victoria Peterson <victoriapeterson09@gmail.com>
+#
+# License: BSD (3-clause)
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.pipeline import Pipeline
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+from sklearn.model_selection import ShuffleSplit, cross_val_score
+
+from mne import (Epochs, pick_types, events_from_annotations,
+                 EvokedArray)
+from mne.channels import make_standard_montage
+from mne.io import concatenate_raws, read_raw_edf
+from mne.datasets import eegbci
+from mne.decoding import CSP
+from mne.decoding import SSD
+
+###############################################################################
+# Set parameters and read data
+
+# avoid classification of evoked responses by using epochs that start 1s after
+# cue onset.
+tmin, tmax = -1., 4.
+event_id = dict(left=2, right=3)  # Motor imagery: left vs right hand
+subject = 2
+runs = [4, 8, 12]
+
+raw_fnames = eegbci.load_data(subject, runs)
+raw = concatenate_raws([read_raw_edf(f, preload=True) for f in raw_fnames])
+eegbci.standardize(raw)  # set channel names
+montage = make_standard_montage('standard_1005')
+raw.set_montage(montage)
+
+# strip channel names of "." characters
+raw.rename_channels(lambda x: x.strip('.'))
+# for the sake of comparison, we save the raw object before filteing
+raw_ssd = raw.copy()
+##############################################################################
+# Data filtering.
+
+# we are going to filter data in the alpha band
+freq_signal = [8, 12]
+# apply band-pass filter
+raw.filter(freq_signal[0], freq_signal[1], fir_design='firwin',
+           skip_by_annotation='edge')
+
+events, _ = events_from_annotations(raw, event_id=dict(T1=2, T2=3))
+
+picks = pick_types(raw.info, meg=False, eeg=True, stim=False, eog=False,
+                   exclude='bads')
+
+# Extract epochs between 1 and 2s
+epochs = Epochs(raw, events, event_id, tmin, tmax, proj=True, picks=picks,
+                baseline=None, preload=True)
+epochs_cropped = epochs.copy().crop(tmin=1., tmax=2.)
+labels = epochs.events[:, -1] - 2
+epochs_data = epochs_cropped.get_data()
+###############################################################################
+# Traditional CSP+LDA pipeline
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+# define pipelines with monte-carlo simulations
+cv = ShuffleSplit(5, test_size=0.2, random_state=42)
+# pipeline methods
+lda = LinearDiscriminantAnalysis()
+csp = CSP(n_components=4, reg=None, log=True, norm_trace=False)
+pipe_csp = Pipeline([('CSP', csp), ('LDA', lda)])
+# this is the accuracy we would like to improve by adding SSD into the pipeline
+scores_csp = cross_val_score(pipe_csp, epochs_data, labels, cv=cv,
+                             n_jobs=1).mean()
+###############################################################################
+# SSD enhances SNR, thus what is considered 'noise' should be defined
+# Typically the 2 Hz sourronunding frequencies are taken
```

Comment:
```suggestion
# Typically bands of 2 Hz surrounding the frequencies of interest are taken
```
This needs to be reworded, I am not sure what is meant. I inserted what I think it means - but might be wrong.